### DISCUSSION TITLE
Calculate c4CountryCode by receiver peppol id scheme

### DISCUSSION THREAD

**Discussion by francescodiperna** [phax/peppol-reporting#11] [General]
Hi, there is a way to calculate the country of the receiver using the peppol id scheme?
For istance, the italian scheme are 0201, 0211, 0210 and in this case the country code is IT. Having the list of all scheme and the mapping to to country code is easy to implement a method for calculating the contry code.
Do you have implemented something like this?

Bye

**Comment by phax** [comment]
Hi @francescodiperna , no unfortunately there isn't a secure way to do it. If such a way would be possible, we would have proposed it ourselves. The main problems are:
* Some participant identifier schemes are "international" and you can't deduce the effective country
* Other schemes (like VAT numbers) look "national", but are not. E.g. Chinese companies having a Polish VAT number (real world scenario btw.)

You must always use the country code of the legal entity of the End User. Based on the KYC (Know Your Customer) Principles that you need to follow in OpenPeppol (see the Internal Regulations), you must have that in your masterdata somewhere.

**Reply by francescodiperna** [reply]
Clear, thanks
